concent_api/core/message_handlers.py

```diff
@@ -820,7 +820,7 @@ def handle_send_force_payment(
 
     # Concent defines time T2 (end time) equal to youngest payment_ts from passed SubtaskResultAccepted messages from
     # subtask_results_accepted_list.
-    payment_ts = min(
+    youngest_payment_ts = max(
```

Comment:
Yeah, this is a bug and a pretty serious one. It allows the provider to run the settlement algorithm as many times as there are acceptances, getting paid for **all** the acceptances each time.

The fix looks good. If you had a separate pull request, but I can't because I have not looked at the rest of this pull request.

Reply:
Oh, actually it's not that bad. It's the timestamp in the message so that most it can do is to confuse the provider. I confused it for a moment with `closure_time` in the settlement payment.

Still, it's a bug. Good catch.

Please make sure that `closure_time` in the payment is exactly the same as this `payment_ts`. Actually, I may be a good idea to move the calculation of this timestamp to bankster to have it in one place.

Reply:
Please rephrase the comment above by removing `T2` from it, as we no longer use it anywhere in the code

Reply:
> Please rephrase the comment above by removing `T2` from it, as we no longer use it anywhere in the code.

I did use it also in next commit, where I add function in bankster. I would like to leave it. It can be helpful for someone who know documentation.

Reply:
Done #1020